TAF_EasyRest/TestFramework/PageComponents/Moderator/UnapprovedRestaurantsPageComponent .cs

```diff
@@ -0,0 +1,37 @@
+﻿namespace TestFramework.PageComponents.Moderator
+{
+    public class UnapprovedRestaurantsPageComponent
+    {
+        private IWebDriver driver;
+
+        public UnapprovedRestaurantsPageComponent(IWebDriver driver)
+        {
+            this.driver = driver;
+        }
+
+        private IReadOnlyCollection<IWebElement> _restaurantsGrid => driver.FindElements(By.XPath("//div[contains(@class, 'Grid-item')]"));
+
+
+
+        private IWebElement _approveButton => driver.FindElement(By.XPath("//span[text() = 'Approve']/parent::button"));
+
+        public UnapprovedRestaurantsPageComponent ClickApproveButton()
+        {
+            var firstRestaurantFromGrid = _restaurantsGrid.ElementAt(0);
+            var approveButton = firstRestaurantFromGrid.FindElement(By.XPath(".//span[text() = 'Approve']/parent::button"));
+
+            approveButton.Click();
+
+            return new UnapprovedRestaurantsPageComponent(driver);
```

Comment:
why don't you use "return this" here?

Reply:
Return this - gives me the same elements on the page. But in fact, the list of restaurants is changed. So I have to re-read this list (_restaurantsGrid in my case) and I use new...